bin/concatenate_all.py

```diff
@@ -35,73 +39,83 @@ def main():
         semester = lichen_config_data["semester"]
         course = lichen_config_data["course"]
         gradeable = lichen_config_data["gradeable"]
-        expressions = None
-        if("regex" in lichen_config_data):
-            # this assumes regex is seperated by a ','
-            expressions = lichen_config_data["regex"].split(',')
+
+        # this assumes regex is seperated by a ','
+        expressions = lichen_config_data["regex"].split(',')
```

Comment:
And expressions here is the actual regex pattern?  Variable name could be improved here.

This one should never have a '..' (and maybe some other character restriction, but backwards directory crawling is the big risk)
(Though perhaps in the code below currently does not allow that security risk)

Reply:
We will be checking to make sure that it is one of the allowed directories, we get a collection of all of the files in all of the subdirectories of each of those allowed directories and then apply the regex to those filenames.  Unless I am misunderstanding the documentation, there is no way to select directories, and there is no way to select anything other than one of the files in the selected directory or its subdirectories.

Reply:
If someone wrote "../../../../confidential_information/*.txt" that would be an attempt the get a match outside the directory and should result in an error.  So please add that simple check.

Yes, the current code to walk the directory does not look at files above the current directory, but if we were to switch that to a different (library?) function for regex file matching it could be a security bug.

Reply:
So just add a simple check for the string `..` within the provided regex?

Reply:
yes please, defensive programming.

Reply:
Changes made.